### ISSUE TITLE
Reconsider `configstore` package with our own implementation

### ISSUE THREAD

**Issue by Murderlon** [tus/tus-node-server#440]
[`configstore`](https://www.npmjs.com/package/configstore), according to the package itself:

> Easily load and persist config without having to think about where and how.
> The config is stored in a JSON file located in `$XDG_CONFIG_HOME` or `~/.config`.
Example: `~/.config/configstore/some-id.json`

The default location (which we don't change) is for config files, which really doesn't feel right for storing upload metadata. It's more expected to have them live next to your uploads. I think we could quite easily replace this with our own implementation and get rid of another dependency. Perhaps this should be done as a breaking change before we move out of beta.

Related: #407

**Comment by yuriyyakym** [comment]
Today I was struggling with this. [configstore](https://github.com/yeoman/configstore) tries to save metadata at `/root` path, where my node server does not have permission.

I've tried to use [conf](https://github.com/sindresorhus/conf) instead, which is a modernized version of [configstore](https://www.npmjs.com/package/configstore). It makes it possible to configure [cwd](https://github.com/sindresorhus/conf#cwd). But:
- I could not use this package, since it does not support my module resolution settings and would require project refactoring,
- Has interface mismatch: `configstore` has property `all`, whereas in `conf` this property is named `store`.

For now I've worked it around this way:

```
const filestore = new FileStore({ directory: FILES_DIRECTORY });
Object.assign(filestore.configstore, { path: `${FILES_DIRECTORY}/metadata.json` });
```
